core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -174,11 +173,26 @@ private class SystemFunctionalityListener implements DatabaseEventListener {
 
         @Override
         public void databaseDrop(DatabaseEventContext eventContext) {
-
-            forEachSystemLabel((tx, label) -> {
-                tx.findNodes(label, database.name(), eventContext.getDatabaseName())
-                        .forEachRemaining(Node::delete);
-            });
+            final var dbName = eventContext.getDatabaseName();
+            if (!Objects.equals(dbName, db.databaseName())) {
+                if (!db.isAvailable()) {
+                    userLog.warn(
+                            "Database %s not available, skipping apoc trigger cleanup of database %s.",
+                            db.databaseName(), dbName);
+                    return;
+                }
```

Comment:
We risk leaving triggers laying around for dropped databases. But what else can we do if system db is unavailable?